salt/modules/opkg.py

```diff
@@ -370,15 +374,175 @@ def _parse_reported_packages_from_install_output(output):
 
     return reported_pkgs
 
+def _is_valid_operation_line(output_line, include_download = True):
+    '''
+    Returns whether the given line is a valid opkg operation output line
+
+    include_download: Boolean
+        Specifies whether we should count lines starting with 'Downloading' as operation lines
+    '''
+    if output_line.startswith(' '):
+        return False
 
-def _execute_install_command(cmd, parse_output, errors, parsed_packages):
+    operation_strings = ['Installing', 'Removing', 'Upgrading', 'Downgrading']
+    output_split = output_line.split()
+    if output_split and (output_split[0] in operation_strings or (include_download and output_split[0] == 'Downloading')):
+        return True
+
+    return False
+
+def _get_total_packages(cmd):
+    '''
+    Runs the given command in test-mode (--noaction) and returns the number of packages
+    that have to be installed/removed/upgraded/downgraded
+    '''
+    test_cmd = copy.deepcopy(cmd)
+    test_cmd.insert(2, '--noaction')
+    number_of_packages = 0
+
+    out = _call_opkg(test_cmd)
+    if out['retcode'] != 0:
+        return -1
+
+    output_operations = out['stdout'].split('\n')
+    for output in output_operations:
+        if _is_valid_operation_line(output, False):
+            number_of_packages += 1
+
+    return number_of_packages
+
+def _get_operation_from_output(output):
+    '''
+    Gets the current opkg operation from output
+    Example:
+        'Installing vim (17.0.0) on root'
+    Returns: 'install'
+    '''
+    operations = {
+            'Installing': 'install',
+            'Upgrading': 'upgrade',
+            'Removing': 'remove',
+            'Downloading': 'download',
+            'Downgrading': 'downgrade'
+            }
+    output_split = output.split()
+    if not output_split:
+        return None
+
+    opkg_operation = output_split[0]
+    if opkg_operation in operations:
```

Comment:
I think you can do  return operations.get(output_split[0], None) - instead of 4 lines